src/main/java/com/github/firmwehr/gentle/firm/optimization/ArithmeticOptimization.java

```diff
@@ -52,6 +57,8 @@
 		.addStep(ArithmeticOptimization::divByOne,
 			(match, graph, block) -> replace(match.div(), match.mem(), match.other()))
 		.addStep(ArithmeticOptimization::addWithZero, (match, graph, block) -> exchange(match.add(), match.any()))
+		.addStep(ArithmeticOptimization::addSame, (match, graph, block) -> exchange(match.add(),
+			graph.newMul(block, match.val(), newConst(graph, 2, match.add().getMode()))))
```

Comment:
I can't follow the rationale behind rewriting an add with a multiplication, which is going to be way more expensive

Reply:
Multiplication is more general than shifts, so depending on the surrounding context, it would need to rewrite the shift to a multiplication again later - both isn't perfect, but our current overall approach doesn't allow better.